core/src/main/java/de/danoeh/antennapod/core/sync/gpoddernet/GpodnetService.java

```diff
@@ -42,24 +42,58 @@
 import java.util.List;
 import java.util.Locale;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 /**
  * Communicates with the gpodder.net service.
  */
 public class GpodnetService implements ISyncService {
     public static final String TAG = "GpodnetService";
     public static final String DEFAULT_BASE_HOST = "gpodder.net";
-    private static final String BASE_SCHEME = "https";
-    private static final int PORT = 443;
     private static final int UPLOAD_BULK_SIZE = 30;
     private static final MediaType TEXT = MediaType.parse("plain/text; charset=utf-8");
     private static final MediaType JSON = MediaType.parse("application/json; charset=utf-8");
-    private final String baseHost;
+    private String baseScheme;
+    private String baseHost;
+    private int basePort;
+
     private final OkHttpClient httpClient;
     private String username = null;
 
-    public GpodnetService(OkHttpClient httpClient, String baseHost) {
+    // split into schema, host and port - missing parts are null
+    private static Pattern urlsplit_regex = Pattern.compile("(?:(https?)://)?([^:]+)(?::(\\d+))?");
+
+    public GpodnetService(OkHttpClient httpClient, String baseHosturl)  {
         this.httpClient = httpClient;
-        this.baseHost = baseHost;
+
+        Matcher m = urlsplit_regex.matcher(baseHosturl);
+        if (m.matches()) {
+            this.baseScheme = m.group(1);
+            this.baseHost = m.group(2);
+            if (m.group(3) == null) {
+                this.basePort = -1;
+            } else {
+                this.basePort = Integer.parseInt(m.group(3));    // regex -> can only be digits
+            }
+        } else {
+            // no match
+            this.baseScheme = "https";
+            this.baseHost = baseHosturl;    // use original (malformed) URL -> this will cause an exception on connect
```

Comment:
This sounds like it is always malformed. I would say just remove the comment :)

Reply:
I reworded the comment, as it relies on something non-obvious outside the scope of this function

Reply:
I think it would be better to just write something like:

    // URL does not match regex: try to interpret it as a hostname

Reply:
The regex matches:
- domain.com
- http://domain.com
- http://domain.com:port
- https://domain.com
- https://domain.com:port

Anything that doesn't match these (and therefore goes into the else branch) is by definition malformed.
By storing it into `hostname` the connection routine **will** throw an exception. This is not elegant but it is the way this app has handled it before.

The fact that malformed URLs are handled by the connection routine is non-obvious. So I would like to put that info into the comment, as well as the fact that the URL at that point is malformed. Something like this:

`// URL did not match regex, i.e. it is malformed: use it anyway -> this will cause an exception on connect`

Your thoughts

Reply:
Ooh, okay. Sorry. I wasn't aware that the regex already captures the hostname

Reply:
Yes. It looks for a `http://` or `https://` at the beginning of the string (if present) and a `: ` followed by digits (!) at the end (if present). The remaining strings is considered to be the hostname.

It is only a simplified URL de-constructor, but it is enough to give the user the opportunity to point AntennaPod to a gpodder.net instance running on her/his own local server.

Do you want me to change the comment?

**Results of the regex match**

```
URL domain.com
==>
scheme: null
host: domain.com
port: null

URL http://domain.com
==>
scheme: http
host: domain.com
port: null

URL http://domain.com:1234
==>
scheme: http
host: domain.com
port: 1234

URL https://domain.com
==>
scheme: https
host: domain.com
port: null

URL https://domain.com:1234
==>
scheme: https
host: domain.com
port: 1234

URL https://domain.com:abc
==>
no match (port number must be digits)

URL xxx://domain.com
==>
no match (schema is neither `http` nor `https`)
```

Reply:
> Do you want me to change the comment?

No - all fine :)